### PR TITLE
SONARIAC-1186 Support "toYaml" function in Helm template evaluation

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: sonar-helm-for-iac/helm-imports.go [hunk 1]
```diff
@@ -268,3 +280,65 @@ func required(warningMessage string, value interface{}) (interface{}, error) {
 func emptyText(_ string) string {
 	return ""
 }
+
+func toYaml(input interface{}) string {
```
Comment (jonas-wielage-sonarsource) [inline sonar-helm-for-iac/helm-imports.go:284]: The functions all look good!
I'm not 100% familiar with the importing logic of go, but can't we just import this file (https://github.com/helm/helm/blob/c6d2f6dba1dd5fd2e417167b4964e5b0a8bb0beb/pkg/engine/funcs.go) and use the functions from there? Than we wouldn't need to reimplement the logic.

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, please see my questions before merging

File: its/ruling/src/test/java/org/sonarsource/iac/IacRulingTest.java [hunk 2]
```diff
@@ -148,6 +148,7 @@ private void run_ruling_test(String project, Map<String, String> projectProperti
       .setProperty("sonar.scm.disabled", "true")
       .setProperty("sonar.internal.analysis.failFast", "true")
       .setProperty("sonar.project", project)
+      .setProperty("sonar.verbose", "true")
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply its/ruling/src/test/java/org/sonarsource/iac/IacRulingTest.java:151]: Good catch. It was a mistake

File: its/ruling/src/test/resources/sources/kubernetes/helm/milestone/L1M3/templates/pod.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 apiVersion: v1
 kind: Pod
-{{ toYaml .Values.metadataPod }} # TODO SONARIAC-1184 HelmEvaluator evaluateTemplate fails for default and toYaml function
+metadata:
+  {{ toYaml .Values.metadataPod }}
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply its/ruling/src/test/resources/sources/kubernetes/helm/milestone/L1M3/templates/pod.yaml:4]: The `metadata` tag was missing, so `KubernetesFilePredicate` ignored the file. I will add a second line.

File: sonar-helm-for-iac/helm-imports.go (hunk 1 above)
Comment (mstachniuk) [reply sonar-helm-for-iac/helm-imports.go:284]: We don't have Helm as our dependency

File: its/ruling/src/test/resources/sources/kubernetes/helm/milestone/L1M3/templates/pod.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply its/ruling/src/test/resources/sources/kubernetes/helm/milestone/L1M3/templates/pod.yaml:4]: Oh...but that's a good limitation of our Helm Analysis. We should write this down somewhere to be aware of it.

**Pull request merged**